### PR TITLE
Export `unstable_serialize` from SWR

### PR DESCRIPTION
**Pull request by shuding**
SWR handles key serialization so it can just pass a string to the cache provider. This makes all the interfaces simpler. However it makes getting data from the cache (or mutating a specific key) tricky, because the serialization util isn't exposed as an API.

Previously we had #1257 to expose the serialization of the `useSWRInfinite` key. This PR unifies the API name, as well as exposing the `serialize` API of `useSWR`.

Related: #1324, #1257.

### CODE REVIEW

File: infinite/index.ts [hunk 1]
```diff
@@ -11,18 +11,18 @@ import useSWR, {
...
 
 const INFINITE_PREFIX = '$inf$'
 
 const getFirstPageKey = (getKey: KeyLoader<any>) => {
-  return serialize(getKey ? getKey(0, null) : null)[0]
+  return serializeKey(getKey ? getKey(0, null) : null)[0]
 }
 
-export const getInfiniteKey = (getKey: KeyLoader<any>) => {
+export const serialize = (getKey: KeyLoader<any>) => {
```
Comment (huozhi) [inline infinite/index.ts:25]: Do you think we should return an array here containing all the custom prefixed keys like `$ctx`, `$len` stuff?
That would be easier to access any of them

Comment (shuding) [reply infinite/index.ts:25]: Not for now. I'm actually worried about exposing too many implementation details, we should do enough research about use cases (e.g. why do they need to access the infinite context?) before thinking about exposing those.

**Review by huozhi — APPROVED** [review]

File: infinite/index.ts (hunk 1 above)
Comment (huozhi) [reply infinite/index.ts:25]: I feel the different return type is also bit confusing for new uers (personally), but it's safe to carry on. We can revisit it later.

Would be nice if explicit return type is declared here

Comment (shuding) [reply infinite/index.ts:25]: Oh that's a good point. We should probably only return the data key from `serialize`.

Comment (huozhi) [reply infinite/index.ts:25]: unstable_serialize?

**Review by huozhi — APPROVED** [review]
LGTM

**Comment by piotr-cz** [comment]
This is helpful, however there is no way to retrieve `errorKey` and `isValidatingKey` values.

I'm using [persistent storage](https://swr.vercel.app/docs/advanced/cache#localstorage-based-persistent-cache) for cache and once an item is removed, I'd like to remove it completely from persistent cache as well.

At this moment I'm using:

```js
export function deleteCacheEntry(cache: Cache, key: string): void {
  const keys = [
    key,
    '$err$' + key, // errorKey
    '$req$' + key, // isValidatingKey
  ]

  keys.forEach(key =>
    cache.delete(key)
  ))
}
```

One prefixes change, such code won't have any effect.

**Comment by shuding** [comment]
I think one way to simplify all this is, using only 1 key for all states related to that resource and keep those states as object values:

```js
// old
cache.set(key, data)
cache.set(errorKey, error)
cache.set(validatingKey, isValidating)

// new
cache.set(key, { data, error, isValidating })
```

This will surely be a breaking change but it would be great to add it.

**Comment by piotr-cz** [comment]
@shuding good idea

**Pull request merged**